Add fundrawtransaction

dagurval opened the pull request:
Adds fundrawtransaction RPC call. Includes the following from Bitcoin Core:
https://github.com/bitcoin/bitcoin/pull/6088 - fundrawtransaction
https://github.com/bitcoin/bitcoin/pull/6417 - [QA] fix possible reorg issue in (fund)rawtransaction(s).py RPC test
https://github.com/bitcoin/bitcoin/pull/6444 - Exempt unspendable transaction outputs from dust checks
https://github.com/bitcoin/bitcoin/pull/6415 - Implement watchonly support in fundrawtransaction
https://github.com/bitcoin/bitcoin/pull/6828 - [rpc-tests] fundrawtransaction: Update fee after minRelayTxFee increase
commit bebe58 from https://github.com/bitcoin/bitcoin/pull/7296
https://github.com/bitcoin/bitcoin/pull/7506 - Use CCoinControl selection in CWallet::FundTransaction
https://github.com/bitcoin/bitcoin/pull/7518 - Add multiple options to fundrawtransaction
https://github.com/bitcoin/bitcoin/pull/7967 - [RPC] add feerate option to fundrawtransaction

Includes the updates up until their test framework supported python3. There are more changes that should be ported over after this.

dgenr8 commented:
Awesome!  With this we can supporting atomic tools like atomic swaps https://github.com/decred/atomicswap/pull/37

dgenr8 commented:
Can 86ce0cc8e9f700a7e33b7881b6383f58d6766ff0 be removed?

dagurval commented:
* Removed 86ce0cc
* Rebased
* Fixed a couple of signed/unsigned comparison warnings

@dgenr8 # 6828 is included as commit fef887a, bebe58 from # 7296 is included as commit 40812c6

dgenr8 commented:
Ok, the file was collapsed when I happened to search for that code.
